lib/vyasa_web/components/contexts/read.ex

```diff
@@ -484,12 +484,15 @@ defmodule VyasaWeb.Context.Read do
     socket
   end
 
-  # currently naive hd lookup can be filter based on active toggle,
-  # tree like sheafs can be used to store nested collapsible topics (personal mark collection e.g.)
-  # currently marks merged in and swapped out probably can be singular data structure
-  # managing of lifecycle of marks
-  # if sangh_id is active open
-  defp sync_draft_reflector(%{assigns: %{session: %{sangh: %{id: sangh_id}}}} = socket) do
+  # Allows us to get a reflection of the internal sangh session state and store it within
+  # this component's state.
+  # Currently, we shall do a naive hd lookup on the sheafs within the session.
+  # We could filter the sheaf based on the active flag,
+  # NOTE:
+  # Tree like sheafs can be used to store nested collapsible topics (personal mark collection e.g.)
+  # TODO: @ks0m1c combine the state handling for marks and sheaf by using the marks within the sheaf.
```

Comment:
@ks0m1c this too, looks like the marks state was intentionally half-wired up because the order of new marks is not being updated properly @ mark-creation

![image](https://github.com/user-attachments/assets/67d597f0-4eed-45b8-b07e-78798da56204)

Reply:
I will arrive at a generic to solve this ordering issue in both the creation and updating context